task_sdk/src/airflow/sdk/definitions/taskgroup.py

```diff
@@ -270,7 +272,7 @@ def group_id(self) -> str | None:
     @property
     def label(self) -> str | None:
         """group_id excluding parent's group_id used as the node label in UI."""
-        return self._group_id
+        return self.group_display_name or self._group_id
```

Comment:
@uranusjr I remember you had in your mind when we created the display name for DAG and Task that we consolidate on node class level / base class. Is it something with TaskSDK that we should follow-up on?